frontend/packages/dev-console/src/components/import/app/ApplicationSelector.tsx

```diff
@@ -37,10 +41,12 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({
   useFormikValidationFix(selectedKey.value);
 
   const onDropdownChange = (key: string, application: string) => {
-    setFieldValue('application.selectedKey', key);
-    setFieldTouched('application.selectedKey', true);
-    setFieldValue('application.name', sanitizeApplicationValue(application, key));
-    setFieldTouched('application.name', true);
+    const selField = subPath ? `${subPath}.application.selectedKey` : 'application.selectedKey';
+    const nameField = subPath ? `${subPath}.application.name` : 'application.name';
+    setFieldValue(selField, key);
+    setFieldTouched(selField, true);
```

Comment:
```suggestion
    const nameField = subPath ? `${subPath}.application.name` : 'application.name';
    setFieldValue(selectedKey.name, key);
    setFieldTouched(selectedKey.name, true);
```

Reply:
Since we are using `useField` in the top of this component, the first arugment `selectedKey` will contain Field attributes like `name, value` properties in it. we can make use of it instead of forming the string everytime.